autohooks/settings.py

```diff
@@ -70,7 +70,9 @@ def write(self, filename: Path) -> None:
         overridden.
         """
         if filename.exists():
-            toml_doc = tomlkit.loads(filename.read_text())
+            toml_doc: Union[
```

Comment:
Some errors occure because of this typing:

`autohooks/settings.py:81: error: Unsupported right operand type for in ("Union[Item, Container]")  [operator]
autohooks/settings.py:82: error: Unsupported target for indexed assignment ("Union[Item, Container]")  [index]
autohooks/settings.py:89: error: Value of type "Union[Item, Container]" is not indexable  [index]
autohooks/settings.py:89: error: Item "Item" of "Union[Any, Item, Container]" has no attribute "update"  [union-attr]`

So there seems to be an problem with the typing Union[TOMLDocument, Table]. Might anyone be able to give me some insights into what might be missing?

Reply:
I don't understand this part. IMHO it must be `toml_doc: tomlkit.TOMLDocument`

Reply:
Your right. Thats the correct typing. That was an error on my side. The difficult part is that we are calling entries of the TOMLDocument (for example) "tool" (that should be a table) and check if something is "in" it. later we try to call toml_doc["tool"]["autohooks"].update(). In the context of this module "tool" doesn't necessary need to be defined, so mypy is clueless what it could be. Sadly i don't think i can type "tool" in here. Or do you maybe have an idea how i could do that? So my options seem to be to ignore the errors or check the instance of ["tool"] and ["autohooks"] each time we use them.

Reply:
I am really not sure what's the problem. At the end `TOMLDocument` is also a dict and implements the [__contains__](https://docs.python.org/3.10/reference/datamodel.html#object.__contains__) and [__getitem__](https://docs.python.org/3.10/reference/datamodel.html#object.__getitem__) protocols required for both.

Reply:
TOMLDocument is not the problem but rather the entries that are undefined (in terms of typing). At least that's what i'm interpreting in these errors:
```
autohooks/settings.py:81: error: Unsupported right operand type for in ("Union[Item, Container]")  [operator]
autohooks/settings.py:82: error: Unsupported target for indexed assignment ("Union[Item, Container]")  [index]
autohooks/settings.py:89: error: Value of type "Union[Item, Container]" is not indexable  [index]
autohooks/settings.py:89: error: Item "Item" of "Union[Any, Item, Container]" has no attribute "update"  [union-attr]
```
And i don't think that i can type them in this modul. So my solution would be to ignore these errors if there is no other way

```
  if filename.exists():
            toml_doc: tomlkit.toml_document.TOMLDocument = tomlkit.loads(
                filename.read_text()
            )
        else:
            toml_doc = tomlkit.document()

        if "tool" not in toml_doc:
            toml_doc["tool"] = tomlkit.table(is_super_table=True)
81      if "autohooks" not in toml_doc["tool"]:
82          toml_doc["tool"]["autohooks"] = tomlkit.table()

        config_dict = {
            "mode": str(self.mode.get_effective_mode()),
            "pre-commit": sorted(self.pre_commit),
        }

89      toml_doc["tool"]["autohooks"].update(config_dict)

```